src/test/java/com/uid2/shared/store/reader/RotatingKeysetProviderTest.java

```diff
@@ -139,16 +138,34 @@ public void loadContentMultipleEntries() throws Exception {
         Assert.assertEquals(6, count);
 
         // Site 2 can access 1, it can access its own, but not 3
-        Assert.assertTrue(canAccessKey(2, 1));
-        Assert.assertTrue(canAccessKey(2, 2));
-        Assert.assertFalse(canAccessKey(2, 3));
-        //Only 4 can access null list
-        Assert.assertTrue(canAccessKey(4,4));
-        Assert.assertFalse(canAccessKey(2, 4));
-        //Can still access if there is a duplicate
-        Assert.assertTrue(canAccessKey(1, 5));
-        // Only 6 can access its empty list
-        Assert.assertTrue(canAccessKey(6,6));
-        Assert.assertFalse(canAccessKey(5,6));
+        Assert.assertTrue(canAccessKey(2, 1, MissingAclMode.ALLOW_ALL));
+        Assert.assertTrue(canAccessKey(2, 1, MissingAclMode.DENY_ALL));
+        Assert.assertTrue(canAccessKey(2, 2, MissingAclMode.ALLOW_ALL));
+        Assert.assertTrue(canAccessKey(2, 2, MissingAclMode.DENY_ALL));
+        Assert.assertFalse(canAccessKey(2, 3, MissingAclMode.ALLOW_ALL));
+        Assert.assertFalse(canAccessKey(2, 3, MissingAclMode.DENY_ALL));
+        // null list
+        Assert.assertTrue(canAccessKey(4,4, MissingAclMode.ALLOW_ALL));
+        Assert.assertTrue(canAccessKey(4,4, MissingAclMode.DENY_ALL));
+        Assert.assertTrue(canAccessKey(2, 4, MissingAclMode.ALLOW_ALL));
+        Assert.assertFalse(canAccessKey(2, 4, MissingAclMode.DENY_ALL));
+        Assert.assertTrue(canAccessKey(5, 4, MissingAclMode.ALLOW_ALL));
+        Assert.assertFalse(canAccessKey(5, 4, MissingAclMode.DENY_ALL));
+        // list with duplicates
+        Assert.assertTrue(canAccessKey(1, 5, MissingAclMode.ALLOW_ALL));
+        Assert.assertTrue(canAccessKey(1, 5, MissingAclMode.DENY_ALL));
+        Assert.assertTrue(canAccessKey(2, 5, MissingAclMode.ALLOW_ALL));
+        Assert.assertTrue(canAccessKey(2, 5, MissingAclMode.DENY_ALL));
+        Assert.assertFalse(canAccessKey(4, 5, MissingAclMode.ALLOW_ALL));
+        Assert.assertFalse(canAccessKey(4, 5, MissingAclMode.DENY_ALL));
+        Assert.assertTrue(canAccessKey(5, 5, MissingAclMode.ALLOW_ALL));
+        Assert.assertTrue(canAccessKey(5, 5, MissingAclMode.DENY_ALL));
+        // empty list
+        Assert.assertTrue(canAccessKey(6,6, MissingAclMode.ALLOW_ALL));
+        Assert.assertTrue(canAccessKey(6,6, MissingAclMode.DENY_ALL));
+        Assert.assertTrue(canAccessKey(5,6, MissingAclMode.ALLOW_ALL));
```

Comment:
I think this isn't correct - if a list is empty, it means we don't want to share with anyone, including ID_READER

Reply:
@jon8787 @cody-constine-ttd - The new logic here is:
(1) if a keyset’s allowed_list is null or empty, allow or deny access will be controlled by MissingAclMode enum.
(2) keys belong to the site_id of the keyset, will automatically get access.